metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -687,15 +697,8 @@ public void handleSnapshot(SnapshotReader<ApiMessageAndVersion> reader) {
                             )
                         );
                     }
-                    if (lastCommittedOffset != -1) {
-                        throw new IllegalStateException(
-                            String.format(
-                                "Asked to re-load snapshot (%s) after processing records up to %s",
-                                reader.snapshotId(),
-                                lastCommittedOffset
-                            )
-                        );
-                    }
+
```

Comment:
Can you add a log here indicating that we're processing a snapshot load (and what offset we're loading...)

Reply:
Done.